### DISCUSSION TITLE
Put the website behind a caching server

### DISCUSSION THREAD

**Discussion by DavidJameson** [fltk/fltk#1137] [General]
This library looks VERY interesting - however I found that accessing information on the website was painfully slow.
Why not get a free account at cloudflare (say) so that users can access your site much faster?

**Comment by dannye** [comment]
I'm not a part of the team or anything, but I'll just say that the website does seem very slow right now, but it is usually not slow at all. There must be some issue. This slowness is not the norm.

**Reply by DavidJameson** [reply]
I was just hoping that someone in a position to address this might be watching. I get that it's not always possible to invest in high-end infrastructure, particularly for free stuff, but putting a low powered webserver behind something like Cloudflare will work extremely well

**Reply by dannye** [reply]
I just checked again and it already seems back to normal.

**Reply by dannye** [reply]
Well.. half the time.

@erco77 ?

**Comment by erco77** [comment]
Might be our provider throttling us back because of this:
![image](https://github.com/user-attachments/assets/62285394-1c42-44c5-ba27-641929a35e82)

**Reply by erco77** [reply]
There's been sustained attacks in the last few days that have been slowing the apache server down, (even before being slashdotted today). It's been spotty; sometimes fast, sometimes slow.

Gonna ride it out, and if things don't change I may investigate a caching server, but not looking forward to adding more complexity.

Looking at the server itself when things are slow, there are none of the usual signs of mem/cpu/processes/net traffic overuse. Even when there's lulls in accesses on the server, there'd still be timeouts.

I'm thinking our provider has been throttling us down, as there's really no sign of anything up on the server itself. Usually too much traffic shows up as a memory or cpu hit, and network traffic is usually high, but not this time.

**Comment by DavidJameson** [comment]
The nice thing about using Cloudflare (which really isn’t very 
complicated at all, particularly at the free tier) is that it prevents a 
lot (if not all)  attacks since your real server is hidden and users 
only connect to cloudserver

I was able to set it up for our server in an hour or so (learning as I 
went along) and it was pretty much a set it and forget it configuration.


------ Original Message ------
From "erco77" ***@***.***>
To "fltk/fltk" ***@***.***>
Cc "David Jameson" ***@***.***>; "Author" 
***@***.***>
Date 11/20/2024 4:14:52 PM
Subject Re: [fltk/fltk] Put the website behind a caching server 
(Discussion #1137)

**Reply by erco77** [reply]
Thanks -- might look into it.
